### PR TITLE
Improve documentation and grammar

### PR DESCRIPTION
**Pull request by Inego**

### CODE REVIEW

**Comment by Inego** [comment]
> Great job!
> I've fixed parameter names and `isEmpty` in [8a38d7b](https://github.com/Kotlin/kotlinx.coroutines/commit/8a38d7bfb9f6c4105b4ed2360e5abc962e882f06), we use `[]` to have a proper reference in the generated documentation.
> 
> Will squash all the changes if you don't mind

Now I see. Originally I thought that these parameter references just led to blank pages in the generated documentation, but now I see that these are highlighted with yellow in the signature when clicked. Sorry you had to manually revert all those changes -- you could just tell me to do that :)

File: kotlinx-coroutines-core/common/src/channels/Channel.kt [hunk 1]
```diff
@@ -154,61 +156,61 @@ public interface ReceiveChannel<out E> {
...
+     * received from the channel.
+     * The [select] invocation fails with an exception if the channel
+     * [is closed for `receive`][isClosedForReceive] (see [close][SendChannel.close] for details).
      */
     public val onReceive: SelectClause1<E>
 
     /**
-     * Retrieves and removes the element from this channel suspending the caller while this channel is empty,
-     * or returns `null` if the channel is [closed][isClosedForReceive] without cause
+     * Retrieves and removes an element from this channel if it's not [empty][isEmpty], or suspends the caller while the channel is empty,
```
[the remark this answers is not in the document]
Comment (Inego) [reply kotlinx-coroutines-core/common/src/channels/Channel.kt]: OK, thanks.

File: kotlinx-coroutines-core/common/src/flow/Builders.kt [hunk 2]
```diff
@@ -245,22 +251,22 @@ public fun <T> channelFlow(@BuilderInference block: suspend ProducerScope<T>.()
...
- * The resulting flow is _cold_, which means that [block] is called on each call of a terminal operator
- * on the resulting flow.
+ * The resulting flow is _cold_, which means that [block] is called every time a terminal operator
+ * is applied to the resulting flow.
  *
  * This builder ensures thread-safety and context preservation, thus the provided [ProducerScope] can be used
- * from any context, e.g. from the callback-based API.
- * The resulting flow completes as soon as the code in the [block] and all its children complete.
+ * from any context, e.g. from a callback-based API.
+ * The resulting flow completes as soon as the code in the [block] and all its children completes.
```
[the remark this answers is not in the document]
Comment (Inego) [reply kotlinx-coroutines-core/common/src/flow/Builders.kt]: I probably understood the original wording wrong. I thought that it was a typo in a grammatical structure `complete(code_in(and(block, all_its_children)))`.
If the intended structure really is `complete(and(code_in(block), all_its_children))`, then maybe it's better to reword it? Like omit the word "code": "... as soon as [block] and all its children complete"?

**Comment by qwwdfsad** [comment]
Sorry for the delay, finally back from a long vacation.
Squashed and merged manually to `develop`.

Thanks a lot for improving the documentation!

**Pull request closed without merging**